### PR TITLE
Feature/l10n br coa

### PR DESCRIPTION
**Pull request by gabrielcardoso21**
Conforme discutido no PR https://github.com/OCA/l10n-brazil/pull/832 foi feita a reestruturação dos planos de contas.

### CODE REVIEW

File: l10n_br_coa/README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (renatonlima) [inline l10n_br_coa/README.rst]: Please, add readme folder to replace this file using OCA readme autogenerate tools

File: l10n_br_coa/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (renatonlima) [inline l10n_br_coa/__manifest__.py]: You can remove this line, it's not necessary in Python 3

File: l10n_br_coa/models/account_chart_template.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
```
Comment (renatonlima) [inline l10n_br_coa/models/account_chart_template.py]: You can remove this line, it's not necessary in Python 3

File: l10n_br_coa/models/l10n_br_account_tax_template.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (renatonlima) [inline l10n_br_coa/models/l10n_br_account_tax_template.py]: You can remove this line, it's not necessary in Python 3

File: l10n_br_coa/models/l10n_br_account_tax_template.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# Copyright 2020 KMEE
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class L10nBrAccountTaxTemplate(models.Model):
+
```
Comment (renatonlima) [inline l10n_br_coa/models/l10n_br_account_tax_template.py]: New line not necessary here

File: l10n_br_coa/views/l10n_br_account_tax_template.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (renatonlima) [inline l10n_br_coa/views/l10n_br_account_tax_template.xml:1]: I Think this file can be removed!

File: l10n_br_coa_simple/migrations/12.0.1.0.1/pre-migration.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# Copyright (C) 2020 - TODAY Marcel Savegnago - Escodoo
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    openupgrade.update_module_names(env.cr,
```
Comment (renatonlima) [inline l10n_br_coa_simple/migrations/12.0.1.0.1/pre-migration.py]: Após o comando para renomear o módulo, você precisa atualizar o ir.model.data dos dados de account.tax.group e accounta.tax.template para atualizar do módulo l10n_br_coa_simple para l10n_br_coa.

File: l10n_br_coa/models/l10n_br_account_tax_template.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    _name = 'l10n_br_account.tax.template'
```
Comment (renatonlima) [inline l10n_br_coa/models/l10n_br_account_tax_template.py:8]: Porque não utilizar o account.tax.template? o chart_template_id não é requerido... https://github.com/odoo/odoo/blob/12.0/addons/account/models/chart_template.py#L76

File: l10n_br_coa/models/account_chart_template.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+from odoo import models
+
+
+class AccountChartTemplate(models.Model):
+
+    _inherit = 'account.chart.template'
+
+    def try_loading_for_current_company(self):
+        for template in self.env['l10n_br_account.tax.template'].search([]):
```
Comment (renatonlima) [inline l10n_br_coa/models/account_chart_template.py]: Deveria verificar se a moeda do plano de contas é o BRL, senão qualquer plano que tentar carregar vai sempre adicionar os impostos do l10n_br_coa

**Review by renatonlima — DISMISSED** [review]

File: l10n_br_coa/models/l10n_br_account_tax_template.py (hunk 8 above)
Comment (mileo) [reply l10n_br_coa/models/l10n_br_account_tax_template.py:8]: Creio que vc tenha se confundido 

https://github.com/odoo/odoo/blob/12.0/addons/account/models/chart_template.py#L789-L794

Comment (renatonlima) [reply l10n_br_coa/models/l10n_br_account_tax_template.py:8]: Eh eu confundi com o AccountAccountTemplate

File: l10n_br_account/hooks.py [hunk 10]
```diff
@@ -33,25 +33,3 @@ def pre_init_hook(cr):
                where fiscal_document_line_id IS NULL;""",
         (fiscal_doc_line_id,),
     )
-
-
-def post_init_hook(cr, registry):
```
Comment (mileo) [inline l10n_br_account/hooks.py:38]: Será que isso não deverá ser refatorado para ser aplicado no módulo l10n_br_coa_simple?

**Review by mileo — DISMISSED** [review]

**Review by marcelsavegnago — COMMENTED** [review]
Localmente não percebi problema instalando os módulos l10n_br_fiscal e l10n_br_account tanto com dados demo como sem dados demo. Porém, no runbot está com o generic coa instalado.

**Review by mileo — APPROVED** [review]

**Comment by renatonlima** [comment]
@mileo e @marcelsavegnago,

No travis e no runbot esta sendo carregado por padrão o l10n_coa_generic do core do Odoo na empresa principal, isso se deve a forma como é feito a instalação dos módulos, tanto no travis como no runbot primeiro são instalados todos os módulos do core e as dependências (oca_dependencies.txt) neste momento é instalado o módulo account (do core) e seus dados de demonstração, e a instalação do plano de contas na empresa padrão que esta apenas com os dados de demo do módulo base e neste momento da instalação do account também é carregado o plano de contas e por isso vai carregar o l10n_coa_generic, só depois de carregados estes módulos (de dependências) ai é realizada a instalação dos módulos do repo l10n-brazil com os testes ativados.

Eu quebrei bem a cabeça para perceber isso quando criei o l10n_br_simple, por isso que no teste do l10n_br_simple é criado uma nova empresa e carregado o plano de contas do l10n_br_simple e módulo fiscal criamos uma segunda empresa (do regime simples nacional) para carregar o plano de contas do l10n_br_simple e depois fazer testes com o lançamento contábeis do simples nacional.

**Review by rvalyi — DISMISSED** [review]
LICENSE: @renatonlima and me think **we better license these COA modules under LGPL** rather than AGPL.

1st of all you cannot really license data, so in any case it's borderline and imposing a copyleft license on it seems even more borderline. So it's not useful in its AGPL form: Odoo SA could copy the data files in Odoo EE and nothing would happen.

2nd: we have to use strategy: we know from experience that despite Odoo SA itself moved from GPL to AGPL in 2009 (and back to LGPL in 2015), they never accept depending on AGPL for their business offers like Odoo EE. When they took code from Akretion they always asked us to relicense to LGPL or said they would re-implement otherwise.

Now, we think this is a shot in the feets if Odoo SA re-implements these base modules instead of just recommending the OCA ones. If they do that, migrating from Odoo EE to the OCA will be harder and that means more companies locked in Odoo EE forever and less market for us. Having the very basic modules LGPL and Odoo EE friendly is also a way to fish for OCA newcomers: "hey we used this module from the OCA, it looks so good, let's look at another one, wow! it looks can do even better with the OCA modules than Odoo EE..." That kind of logic.

This being said nobody can accuse Akretion of wanting to play the Odoo EE game. We were among the 1st gold partners to bang the door to Odoo SA in 2015 when their partner contract started being centered on selling proprietary licenses and Akretion is author or contributor of around 10% of the OCA modules worldwide (even on v13) with 95% of them being AGPL...
While we also advocated for changing l10n_br_base to LGPL, we would never advocate moving away from the AGPL license for l10n_br_fiscal or l10n_br_account and later modules...

So in short, we think ensuring a better Odoo EE compatibility for base modules and base master data is an advantage for the OCA because EE users are likely to be willing to move to the OCA after somebody sold them some unrealistic dream on EE while it's very unlikely OCA happy users will want to move away to the EE proprietary world....

What do you think?

**Review by marcelsavegnago — APPROVED** [review]

**Comment by mileo** [comment]
> LICENSE: @renatonlima and me think **we better license these COA modules under LGPL** rather than AGPL.
> 
> 1st of all you cannot really license data, so in any case it's borderline and imposing a copyleft license on it seems even more borderline. So it's not useful in its AGPL form: Odoo SA could copy the data files in Odoo EE and nothing would happen.
> 
> 2nd: we have to use strategy: we know from experience that despite Odoo SA itself moved from GPL to AGPL in 2009 (and back to LGPL in 2015), they never accept depending on AGPL for their business offers like Odoo EE. When they took code from Akretion they always asked us to relicense to LGPL or said they would re-implement otherwise.
> 
> Now, we think this is a shot in the feets if Odoo SA re-implements these base modules instead of just recommending the OCA ones. If they do that, migrating from Odoo EE to the OCA will be harder and that means more companies locked in Odoo EE forever and less market for us. Having the very basic modules LGPL and Odoo EE friendly is also a way to fish for OCA newcomers: "hey we used this module from the OCA, it looks so good, let's look at another one, wow! it looks can do even better with the OCA modules than Odoo EE..." That kind of logic.
> 
> This being said nobody can accuse Akretion of wanting to play the Odoo EE game. We were among the 1st gold partners to bang the door to Odoo SA in 2015 when their partner contract started being centered on selling proprietary licenses and Akretion is author or contributor of around 10% of the OCA modules worldwide (even on v13) with 95% of them being AGPL...
> While we also advocated for changing l10n_br_base to LGPL, we would never advocate moving away from the AGPL license for l10n_br_fiscal or l10n_br_account and later modules...
> 
> So in short, we think ensuring a better Odoo EE compatibility for base modules and base master data is an advantage for the OCA because EE users are likely to be willing to move to the OCA after somebody sold them some unrealistic dream on EE while it's very unlikely OCA happy users will want to move away to the EE proprietary world....
> 
> What do you think?

Favor abrir um issue para discutirmos isso e não mudar o foco do pull resquest
